hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/SlowPeerTracker.java

```diff
@@ -80,7 +80,7 @@ public class SlowPeerTracker {
    * Number of nodes to include in JSON report. We will return nodes with
    * the highest number of votes from peers.
    */
-  private final int maxNodesToReport;
+  private int maxNodesToReport;
```

Comment:
Please set this to `volatile`. Although it doesn't make a big difference here, I think it's better to be consistent with other reconfig changes. What do you think of this?

Reply:
Yes, you have made good point @tomscut, this can be done to remain in line with other reconfig changes, however it might cause bit of a performance issue for JMX metrics API overall, hence I was bit reluctant to make the change. But if you have strong preference, I can make the change, WDYT?

Reply:
This field is almost always read-only and rarely changed, and for read-only operations, it doesn't have much impact. WDYT?

Reply:
Yeah it's fine I think, this is not a big concern either ways. So let me make this change.